### PR TITLE
OSDOCS-7856: Migrated AWS Secrets Manager

### PR DESCRIPTION
**Pull request by EricPonvelle**
Version(s):
`enterprise-4.13+`

Issue:
[OSDOCS-7856](https://issues.redhat.com/browse/OSDOCS-7856)

Link to docs preview:
- [Using AWS Secrets Manager CSI on Red Hat OpenShift on AWS with STS](https://65003--docspreview.netlify.app/openshift-rosa/latest/cloud_experts_tutorials/cloud-experts-aws-secret-manager)

QE review:
- [ ] QE has approved this change.

Additional information:
Migrated the "Using AWS Secrets Manager CSI on Red Hat OpenShift on AWS with STS" topic from the MOBB docs to ROSA docs.

### CODE REVIEW

File: rosa_tutorials/rosa-mobb-aws-secret-manager.adoc [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+$ export OIDC_ENDPOINT=$(oc get authentication.config.openshift.io cluster \
+   -o jsonpath='{.spec.serviceAccountIssuer}' | sed  's|^https://||')
+$ export AWS_ACCOUNT_ID=`aws sts get-caller-identity --query Account --output text`
+$ export AWS_PAGER=""
+----
+
+[id="rosa-mobb-aws-secret-manager-deply-aws-secrets"]
+== Deploy the AWS Secrets and Configuration Provider
+
+. Use Helm to register the secrets store csi driver:
```
Comment (bhardesty) [inline rosa_tutorials/rosa-mobb-aws-secret-manager.adoc:76]:
```suggestion
. Use Helm to register the secrets store CSI driver:
```

File: rosa_tutorials/rosa-mobb-aws-secret-manager.adoc [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+----
+
+. Update your Helm repositories:
++
+[source,terminal]
+----
+$ helm repo update
+----
+
+. Install the secrets store csi driver:
```
Comment (bhardesty) [inline rosa_tutorials/rosa-mobb-aws-secret-manager.adoc:91]:
```suggestion
. Install the secrets store CSI driver:
```

File: rosa_tutorials/rosa-mobb-aws-secret-manager.adoc [hunk 3]
```diff
@@ -0,0 +1,334 @@
...
++
+[source,terminal]
+----
+$ ROLE_ARN=$(aws iam create-role --role-name openshift-access-to-mysecret \
+    --assume-role-policy-document file://trust-policy.json \
+    --query Role.Arn --output text)
+$ echo $ROLE_ARN
+----
+
+. Attach role to the policy:
```
Comment (bhardesty) [inline rosa_tutorials/rosa-mobb-aws-secret-manager.adoc:201]:
```suggestion
. Attach the role to the policy:
```

File: rosa_tutorials/rosa-mobb-aws-secret-manager.adoc [hunk 4]
```diff
@@ -0,0 +1,334 @@
...
++
+[source,terminal]
+----
+$ oc exec -it my-application -- cat /mnt/secrets-store/MySecret
+----
+
+[id="rosa-mobb-aws-secret-manager-cleanup"]
+== Clean up
+
+. Delete application:
```
Comment (bhardesty) [inline rosa_tutorials/rosa-mobb-aws-secret-manager.adoc:287]:
```suggestion
. Delete the application:
```

File: cloud_experts_tutorials/rosa-mobb-aws-secret-manager.adoc [hunk 5]
```diff
@@ -0,0 +1,334 @@
...
+  - name: my-application-deployment
+    image: k8s.gcr.io/e2e-test-images/busybox:1.29
+    command:
+      - "/bin/sleep"
+      - "10000"
+    volumeMounts:
+    - name: secrets-store-inline
+      mountPath: "/mnt/secrets-store"
+      readOnly: true
+EOF
```
Comment (michaelryanmcneill) [inline cloud_experts_tutorials/rosa-mobb-aws-secret-manager.adoc:274]: This fails due to a pod security admission controller. We need to update this deployment to address that (and we need to update all of our secrets store CSI docs to do that). Will circle back on this one.

**Review by michaelryanmcneill — CHANGES_REQUESTED** [review]
Made some small changes, but looks like we've got an issue to fix on our test deployment. I'll circle back with a fix for that tomorrow.

File: cloud_experts_tutorials/rosa-mobb-aws-secret-manager.adoc [hunk 6]
```diff
@@ -0,0 +1,333 @@
...
+    system:serviceaccount:csi-secrets-store:secrets-store-csi-driver
+$ oc adm policy remove-scc-from-user privileged \
+    system:serviceaccount:csi-secrets-store:csi-secrets-store-provider-aws
+----
+
+. Delete the AWS provider:
++
+[source,terminal]
+----
+$ oc -n csi-secrets-store delete -f \
```
Comment (michaelryanmcneill) [inline cloud_experts_tutorials/rosa-mobb-aws-secret-manager.adoc:314]: The link needs to be updated:

```
oc -n csi-secrets-store delete -f \
    https://raw.githubusercontent.com/rh-mobb/documentation/main/content/misc/secrets-store-csi/aws-provider-installer.yaml
```

File: cloud_experts_tutorials/rosa-mobb-aws-secret-manager.adoc [hunk 7]
```diff
@@ -0,0 +1,333 @@
...
+----
+
+. Deploy the AWS provider:
++
+[source,terminal]
+----
+$ oc -n csi-secrets-store apply -f \
+    https://raw.githubusercontent.com/rh-mobb/documentation/main/content/misc/secrets-store-csi/aws-provider-installer.yaml
+----
+
```
Comment (michaelryanmcneill) [inline cloud_experts_tutorials/rosa-mobb-aws-secret-manager.adoc:106]: Insert a new step here:

Label the Secrets Store CSI Driver to allow use with the restricted pod security profile:

```
$ oc label csidriver.storage.k8s.io/secrets-store.csi.k8s.io security.openshift.io/csi-ephemeral-volume-profile=restricted
```

File: cloud_experts_tutorials/rosa-mobb-aws-secret-manager.adoc (hunk 5 above)
Comment (michaelryanmcneill) [reply cloud_experts_tutorials/rosa-mobb-aws-secret-manager.adoc:273]: This is fixed with the changes above.

**Review by michaelryanmcneill — CHANGES_REQUESTED** [review]
We figured out the problem with the deployment (applies to 4.13+). Fixed in the below comment. Ping me once you've gotten it updated @EricPonvelle and I'll re-review to make sure all works as intended.

File: cloud_experts_tutorials/rosa-mobb-aws-secret-manager.adoc (hunk 5 above)
Comment (EricPonvelle) [reply cloud_experts_tutorials/rosa-mobb-aws-secret-manager.adoc:280]: @michaelryanmcneill So to clarify, I do not need to change anything from the highlighted lines 249 to 279?

Comment (michaelryanmcneill) [reply cloud_experts_tutorials/rosa-mobb-aws-secret-manager.adoc:280]: Correct!

File: cloud_experts_tutorials/rosa-mobb-aws-secret-manager.adoc (hunk 6 above)
Comment (michaelryanmcneill) [reply cloud_experts_tutorials/rosa-mobb-aws-secret-manager.adoc:321]: @EricPonvelle this was missed! Other than that, we're good to merge.

**Pull request merged**